### DISCUSSION TITLE
How to resolve commands from a string?

### DISCUSSION THREAD

**Discussion by stecydube** [SilentVoid13/Templater#634] [Help]
First, a simple setup:

### MyTemplate.md
```js
<%*
    const {SomeClass} = customJS;
    tR += SomeClass.getContent();
%>
```

### SomeClass.js
```js
class SomeClass {
  getContent () {
    // As a an example, can be any template command
    return `<% tp.date.yesterday("YYYY-MM-DD") %>`;
  }
}
```

## Templater setup
- Enable folder templates
- Add a new folder template for `/` with "MyTemplate.md"

## Behavior

Create a new file in root and Templater will see this and run the MyTemplate template.

Obviously the result of `getContent` is "pasted" as is in the file, so the "Untitled" note will contain:
```
<% tp.date.yesterday("YYYY-MM-DD") %>
```

## What I want

The "Untitled" note would contain:
```
2022-05-10
```

## I tried

Using `tp.file.include` does not work for provided content as the method expects a file (either a path or a TFile)

**Comment by welpdx** [comment]
I tried your set up and couldn't get it to work:
1. I get an error saying ![img](https://i.imgur.com/FLivweg.jpeg). I couldn't get your basic setup right. Shouldn't the `.js` file export a function? See [Templater Wiki](https://silentvoid13.github.io/Templater/user-functions/script-user-functions.html#define-a-script-user-function). And then import the function via `tp.user.fileName()`?
2. For `tp.file.include()`, make sure to add the `await` keyword in front of it since the function is async and returns a promise. So do: `tR+=await tp.file.include("[[example]]")`.
Hope that helps.

~welp

**Reply by stecydube** [reply]
You need the CustomJS plugin as the script is made for it.

**Reply by welpdx** [reply]
Oh ok, gotcha. I knew I was missing something lol. 

Ok, I got your set up working by removing the script from templater. See here? 
![img](https://i.imgur.com/S98PAXj.jpeg).

You don't want that. You want customJS as the only plugin that uses .js file. This is why the error occurs because templater sees the .js file and is trying to use it as a [templater User Script](https://silentvoid13.github.io/Templater/user-functions/script-user-functions.html#define-a-script-user-function). 

Your setup worked for me when I added my SomeClass.js script in a diff folder solely for customJS.

**Comment by AB1908** [comment]
When running includes, would it make sense to pass in the `tp` object and run commands on that instead of trying to resolve strings? My guess would be no but wanted to throw that out there anyway.
